### PR TITLE
Iter9

### PR DESCRIPTION
**Pull request by ajugalushkin**

### CODE REVIEW

File: internal/handler/handler.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+type Handler struct {
+	servAPI *service.Service
+	cfg     *config.Config
+}
+
+func NewHandler(servAPI *service.Service, cfg *config.Config) *Handler {
+	return &Handler{
+		servAPI: servAPI,
+		cfg:     cfg}
```
Comment (arttet) [inline internal/handler/handler.go]: CHANGE REQUEST: please format your code.

File: internal/handler/handler.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
+package handler
+
+import (
+	"fmt"
+	"github.com/ajugalushkin/url-shortener-version2/internal/config"
```
Comment (arttet) [inline internal/handler/handler.go:9]: CHANGE REQUEST: please sort imports

File: internal/handler/handler.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+func (s Handler) HandleSave(context echo.Context) error {
+	if context.Request().Method != http.MethodPost {
+		logger.Log.Debug("Wrong type request",
+			zap.String("status", strconv.Itoa(http.StatusBadRequest)),
+			zap.String("size", strconv.Itoa(0)),
+		)
+		return context.String(http.StatusBadRequest, "Wrong type request")
+	}
+
+	body, err := io.ReadAll(context.Request().Body)
```
Comment (arttet) [inline internal/handler/handler.go]: CHANGE REQUEST: please move it to the `validate` package

File: internal/logger/logger.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package logger
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+	"time"
+)
+
+var Log *zap.Logger = zap.NewNop()
```
Comment (arttet) [inline internal/logger/logger.go]: CHANGE REQUEST: please don't use global variables

File: internal/logger/logger.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	zl, err := cfg.Build()
+	if err != nil {
+		return err
+	}
+
+	Log = zl
+	return nil
+}
+
+func RequestLogger(next echo.HandlerFunc) echo.HandlerFunc {
```
Comment (arttet) [inline internal/logger/logger.go]:
```suggestion
func middlewareLogger(next echo.HandlerFunc) echo.HandlerFunc {
```

File: internal/model/json.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package model
```
Comment (arttet) [inline internal/model/json.go]:
```suggestion
package dto
```

File: internal/storage/file_storage.go [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+	"bytes"
+	"errors"
+	"github.com/ajugalushkin/url-shortener-version2/internal/config"
+	"github.com/ajugalushkin/url-shortener-version2/internal/model"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+type FileStorage struct {
```
Comment (arttet) [inline internal/storage/file_storage.go]: CHANGE REQUEST: it shouldn't be a file storage.

**Review by arttet — CHANGES_REQUESTED** [review]

**Comment by ajugalushkin** [comment]
Reopen pull request

**Comment by ajugalushkin** [comment]
Reopen pull request

**Pull request closed without merging**